test/npm-compatibility/npm-behaviour.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+# Spits out the behaviour of npm in various version resolution scenarios.
+set -e
+
+function get_version() {
+  package_name=$1
+  echo $(jq -r ".packages.\"node_modules/$package_name\".version" package-lock.json)
+}
+
+function get_range() {
+  package_name=$1
+  echo $(jq -r ".dependencies.\"${package_name}\"" package.json)
+}
+
+function test_command() {
```

Comment:
So we never have to puzzle over the behaviour again!

Reply:
My only concern here is maintainability, and specifically two questions:

1. We do have tests that are specifically tied to current versions of packages, but we are trying to keep that to a minimum / under control. Does this new testing introduced provide more packages that potentially risk updating in future and changing our test locks?
2. This script will likely not be maintained. Is the idea that it is only run sporadically? Should we maintain this in the chompfile?

Reply:
Each of the test cases (the subdirectories) uses old versions of their packages, and I've deliberately chosen packages with a single dependency so they should never have test-breaking changes (assuming the old versions aren't changed on `npm` or something weird).

The idea for the script is just to document how I tested `npm`'s behaviour - if you/anyone else wanted to reproduce the data you can run this again, but I think it's unnecessary unless there are breaking changes in `npm`.

Reply:
Thanks for clarifying, that sounds good.